Sources/SwiftSyntaxParser/SyntaxParser.swift

```diff
@@ -92,33 +104,55 @@ public enum SyntaxParser {
   ///
   /// - Parameters:
   ///   - url: The file URL to parse.
+  ///   - languageVersion: Interpret input according to a specific Swift
+  ///       language version number. If `nil`, this inherits the default from
+  ///       the syntax parser library.
+  ///   - enableBareSlashRegexLiteral: Enable or disable the use of forward
+  ///       slash regular-expression literal syntax. If `nil`, this inherits the
+  ///       default from the syntax parser library.
   ///   - diagnosticHandler: Optional callback that will be called for all
   ///       diagnostics the parser emits
   /// - Returns: A top-level Syntax node representing the contents of the tree,
   ///            if the parse was successful.
   /// - Throws: `ParserError`
-  public static func parse(_ url: URL,
-      diagnosticHandler: ((Diagnostic) -> Void)? = nil) throws -> SourceFileSyntax {
+  public static func parse(
+    _ url: URL,
+    languageVersion: String? = nil,
+    enableBareSlashRegexLiteral: Bool? = nil,
```

Comment:
I *feel* like these should be a `Options` object. But I'm not sure how the API should look like, for example
```swift
// array of enums
SyntaxParse.parse(
    url,
    options: [.languageVersion("5.6"), .bareSlashRegexLiteral(true)],
    diagnosticHandler: diagnosticHandler)
// Just a "options" object
SyntaxParse.parse(
    url,
    options: SyntaxParseOptions(languageVersion: "5.1", enableBareSlashRegexLiteral: true),
    diagnosticHandler: diagnosticHandler)
```
The advantage of "options" object is that you can _easily_ pass around and store the object in the clients, when they want to create options at _not_  exact the call site.

WDYT?

Reply:
Alex and I chatted about this and we decided to accept this as is for now until the "language features" proposal is settled.